cppapi/server/tango_const.h.in

```diff
@@ -932,6 +932,7 @@ enum CmdArgType {
 	DEV_ENUM,
 	DEV_PIPE_BLOB,
 	DEVVAR_STATEARRAY,
+	DATA_TYPE_LAST_VALID = DEVVAR_STATEARRAY,
 	DATA_TYPE_UNKNOWN = 100
```

Comment:
Now every time you add a new type you must change two lines. How about:
```cpp
DEVVAR_STATEARRAY,
// Add new types above DATA_TYPE_INVALID.
DATA_TYPE_INVALID,
DATA_TYPE_UNKNOWN
```

And then use `<` instead of `<=`:
```cpp
return (type >= DEV_VOID && type < DATA_TYPE_INVALID) ? CmdArgTypeName[type] : "Unknown"
```

Reply:
@mliszcz , this sounds like a good idea indeed.

> Now every time you add a new type you must change two lines. How about:
> 
> ```c++
> DEVVAR_STATEARRAY,
> // Add new types above DATA_TYPE_INVALID.
> DATA_TYPE_INVALID,
> DATA_TYPE_UNKNOWN
> ```
> [...]

Reply:
@mliszcz @bourtemb If I do that we have two non-valid entries. I also doubt we very often change that structure.

But nevertheless when rethinking the code the additional enum is not required. Instead of having a last valid member entry I'm now computing the last valid index from the array size. This should be more future proof and also cleaner.